### PR TITLE
VC-13884 add semgrep to workflows repo

### PR DESCRIPTION
**Pull request by cburgas-parloa**
This is work on top of VC-13883
This adds semgreps Static Application Security Testing to the mandatory_workflows.yaml. It also streamlines the naming and adds the same conditional output to the trivy workflow.

It also changes two existing workflows, as these were pulling in third-party Actions with a loose versioning, (which Semgrep hates :D )

### CODE REVIEW

File: .github/workflows/mandatory-workflows.yaml [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+        if: always()
+        id: reportlength
+        run: |
+              LL=$(wc -l < vulnerability-report.txt|bc)
+              echo "reportlength=$LL" >> $GITHUB_OUTPUT
+
+      - name: warn if findings
+        if: always() && steps.reportlength.outputs.reportlength > 0
+        run: 
+          LL=$(cat vulnerability-report.txt)
```
Comment (joan-parloa) [inline .github/workflows/mandatory-workflows.yaml:60]: I guess this line is not doing anything?

File: .github/workflows/mandatory-workflows.yaml [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+
+  static-analysis:
+    name: Static Application Security Testing
+    runs-on: ubuntu-latest
+    container:
+      image: returntocorp/semgrep:latest
+
+    steps:
+
+      # step 1
```
Comment (joan-parloa) [inline .github/workflows/mandatory-workflows.yaml:76]: I'd say these comments are not very useful, they add little (and are in fact incorrectly numbered). I'd suggest to remove them.

File: .github/workflows/mandatory-workflows.yaml [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+              LL=$(wc -l < vulnerability-report.txt|bc)
+              echo "reportlength=$LL" >> $GITHUB_OUTPUT
+
+      - name: warn if findings
+        if: always() && steps.reportlength.outputs.reportlength > 0
+        run: 
+          LL=$(cat vulnerability-report.txt)
+          echo "::warning file=vulnerability-report.txt,line=1,col=1::vulnerabilities, secrets, IaC, or licence compliance issues found, either check the vulnerability-report.txt or check the 'echo report if findings' step"
+
+      - name: echo report if findings
```
Comment (joan-parloa) [inline .github/workflows/mandatory-workflows.yaml:63]: I guess `warn if findings` and `echo report if findings` steps could be a single step

File: .github/workflows/mandatory-workflows.yaml [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+      - name: Checkout code
+        uses: actions/checkout@v3
+
+      - name: Run Trivy in GitHub SBOM mode and submit results to Dependency Graph
+        uses: aquasecurity/trivy-action@fbd16365eb88e12433951383f5e99bd901fc618f #v0.12.0
+        with:
+          scan-type: 'fs'
+          format: 'github'
+          output: 'dependency-results.sbom.json'
+          image-ref: '.'
```
Comment (joan-parloa) [inline .github/workflows/mandatory-workflows.yaml:25]: Should it be `scan-ref`?

File: .github/workflows/mandatory-workflows.yaml [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+        uses: aquasecurity/trivy-action@fbd16365eb88e12433951383f5e99bd901fc618f  #v0.12.0
+        with:
+            scan-type: 'fs'
+            format: 'table'
+            output: 'vulnerability-report.txt'
+            image-ref: '.'
+            exit-code: ${{ inputs.EXITCODE }}
+            vuln-type: 'os,library'
+            severity: 'CRITICAL,HIGH'
+            scanners: 'vuln,secret,config,license'
```
Comment (joan-parloa) [inline .github/workflows/mandatory-workflows.yaml:41]: Don't see `license` as a valid input here https://github.com/aquasecurity/trivy-action#customizing

**Review by joan-parloa — COMMENTED** [review]
Mostly nitpicks :)

File: .github/workflows/mandatory-workflows.yaml (hunk 3 above)
Comment (cburgas-parloa) [reply .github/workflows/mandatory-workflows.yaml:63]: merging them

File: .github/workflows/mandatory-workflows.yaml (hunk 5 above)
Comment (cburgas-parloa) [reply .github/workflows/mandatory-workflows.yaml:41]: true, that might be a good OSS contribution :) it is actually handed down to the CLI 
see here https://github.com/parloa/licence-scanning-example/actions/runs/6407525197/job/17394489615

File: .github/workflows/mandatory-workflows.yaml (hunk 4 above)
Comment (cburgas-parloa) [reply .github/workflows/mandatory-workflows.yaml:25]: removing the image-ref gives me a validation error. Should I add the scan-ref additionally?

**Review by ajay-parloa — APPROVED** [review]

**Review by joan-parloa — APPROVED** [review]

**Pull request merged**